examples/multicam/rs-multicam.cpp

```diff
@@ -18,16 +18,21 @@ int main(int argc, char * argv[]) try
 
     std::vector<rs2::pipeline>            pipelines;
 
-    // Start a streaming pipe per each connected device
+    // Capture serial numbers before opening streaming
+    std::vector<std::string>              serials;
     for (auto&& dev : ctx.query_devices())
+        serials.push_back(dev.get_info(RS2_CAMERA_INFO_SERIAL_NUMBER));
+
+    // Start a streaming pipe per each connected device
+    for (auto&& serial : serials)
     {
         rs2::pipeline pipe(ctx);
         rs2::config cfg;
-        cfg.enable_device(dev.get_info(RS2_CAMERA_INFO_SERIAL_NUMBER));
+        cfg.enable_device(serial);
```

Comment:
@dorodnic, it would make this a much simpler change everywhere this change is needed to allow `enable_device()` to accept a `rs2::device` as well as a serial number `string`.  After all, we have `dev` here and the `rs2::config` just wants a device.  Doing the lookup again seems wasteful.  It would be a bigger change to add support for this, but it would make it usage for users much cleaner going forward.
```c++
    for (auto&& dev : ctx.query_devices())
       cfg.enable_device(dev);
```

Reply:
Hi @radfordi 
You are 100% right. I didn't want to expand the scope of this PR beyond simply getting your original work merged, but I also thought about it.